client/controllers/graph.controller.js

```diff
@@ -1,16 +1,85 @@
 (function() {
-    
     "use strict";
     
     angular
-        .module("graph", [])
+        .module("smitty.graph", [])
         .controller("GraphController", GraphController);
     
-    function GraphController($scope, $http){ 
+    GraphController.$inject = ["canvasService"];
+    
+    function GraphController(canvasService) {
+        var vm = this; // View Model
+        
+        var nextVertexId = 0;
```

Comment:
This is better than tagging a vertex-specific ID to the whole view-model. Of course this will probably eventually come from a database anyways so it's not a huge deal, but I think this makes it more readable in the meantime.